### PR TITLE
fix: Use the doc name matched by fuzzy_find_documents when rendering

### LINKED ISSUE #4855: OAuth 2.1 links show OAuth 2.0 (Problem with drafts with version numbers in the name)
### Describe the issue

The links to OAuth 2.1 are now returning the OAuth 2.0 spec. It looks like the `-1` in the URL is being completely ignored. This seems to have changed relatively recently.

https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-07

image

Previously this was showing version -07 of the OAuth 2.1 draft. https://web.archive.org/web/20221109125705/https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-07

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by larseggert**
Fixes #4855

### CODE REVIEW

File: ietf/doc/views_doc.py [hunk 1]
```diff
@@ -812,6 +812,7 @@ def document_html(request, name, rev=None):
     if found.matched_name.startswith('rfc') and name != found.matched_name:
          return redirect('ietf.doc.views_doc.document_html', name=found.matched_name)
 
+    name = found.matched_name
```
[the remark this answers is not in the document]
Comment (larseggert) [reply ietf/doc/views_doc.py:815]: You mean *also* change the conditional?

[the remark this answers is not in the document]
Comment (rjsparks) [reply ietf/doc/views_doc.py:815]: No - if you redirect back with name = found.matched_name, then when you're back in this code name will already be what the current path found.

**Comment by larseggert** [comment]
@rjsparks you mentioned in Slack that this PS is wrong. I tried to do what you said before, i.e.,
> Why aren't you changing the conditional at L812 instead? That would cause the URL to get updated in the browser.

But I guess I misunderstood?

**Comment by larseggert** [comment]
OBE, see #4897.

**Comment by rjsparks** [comment]
I see I have some pending comments on this - failed a submit button somewhere along the way. I'll try to let them come through for completeness, but treat them as FYI only.

**Pull request closed without merging**